pkg/framework/plugins/nodeutilization/normalizer/normalizer.go

```diff
@@ -0,0 +1,159 @@
+// maximum values (included). Values below minimum are rounded up to the
+// minimum value, and values above maximum are rounded down to the maximum
+// value.
+func Clamp[K comparable, N Number, V ~map[K]N](values V, minimum, maximum N) V {
+	return Ceil(Floor(values, minimum), maximum)
+}
+
+// Map applies a function to each element of a map of values. Returns a new
+// slice with the results of applying the function to each element.
+func Map[K comparable, N Number, V ~map[K]N](items []V, fn func(V) V) []V {
+	result := []V{}
+	for _, item := range items {
+		result = append(result, fn(item))
+	}
+	return result
+}
+
+// Negate converts the values of a map to their negated values.
+func Negate[K comparable, N Number, V ~map[K]N](values V) V {
+	result := V{}
+	for key, value := range values {
+		result[key] = -value
+	}
+	return result
+}
+
+// Round rounds the values of a map to the nearest integer. Calls math.Round on
+// each value of the map.
+func Round[K comparable, N Number, V ~map[K]N](values V) V {
+	result := V{}
+	for key, value := range values {
+		result[key] = N(math.Round(float64(value)))
+	}
+	return result
+}
+
+// Floor receives a minimum value and a map of values. Returns a list of
+// values where each value is greater or equal than the minimum value. In
+// other words, this function rounds up the values to the minimum value
+// provided.
+func Floor[K comparable, N Number, V ~map[K]N](values V, limit N) V {
+	result := V{}
+	for key, value := range values {
+		result[key] = N(math.Max(float64(value), float64(limit)))
+	}
+	return result
+}
+
+// Ceil receives a maximum value and a list of values. Returns a list of values
+// where each value is less or equal than the maximum value provided. In other
+// words, this function rounds down the values to the maximum value provided.
+func Ceil[K comparable, N Number, V ~map[K]N](values V, limit N) V {
+	result := V{}
+	for key, value := range values {
+		result[key] = N(math.Min(float64(value), float64(limit)))
+	}
+	return result
+}
+
+// Sum sums up the values of two maps. Values are expected to be of Number
+// type. Original values are preserved. If a key is present in one map but
+// not in the other, the key is ignored.
+func Sum[K comparable, N Number, V ~map[K]N](mapA, mapB V) V {
+	result := V{}
+	for name, value := range mapA {
+		if _, ok := mapB[name]; !ok {
+			continue
+		}
+		result[name] = value + mapB[name]
+	}
+	return result
+}
+
+// Average calculates the average of a set of values. This function receives
+// a map of values and returns the average of all the values. Average expects
+// the values to represent the same unit of measure. You can use this function
+// after Normalizing the values.
+func Average[J, K comparable, N Number, V ~map[J]N](values map[K]V) V {
+	result := V{}
+	for _, imap := range values {
+		for name, value := range imap {
+			result[name] += value
+		}
+	}
+
+	for name := range result {
+		result[name] /= N(len(values))
```

Comment:
`len(values)` currently corresponds to the number of nodes. Though, not every node will have all the resourcenames (here `name`) set. Instead, the function needs to keep a track of how many times each `name` was accessed and divide the total sum at the end by this number.

Reply:
Done.